src/nnet3/nnet-chain-training.cc

```diff
@@ -38,9 +38,11 @@ NnetChainTrainer::NnetChainTrainer(const NnetChainTrainingOptions &opts,
   KALDI_ASSERT(opts.nnet_config.momentum >= 0.0 &&
                opts.nnet_config.max_param_change >= 0.0);
   delta_nnet_ = nnet_->Copy();
+  ScaleNnet(0.0, delta_nnet_);
   bool is_gradient = false;  // setting this to true would disable the
                              // natural-gradient updates.
-  SetZero(is_gradient, delta_nnet_);
+  if (is_gradient)
```

Comment:
we just set is_gradient = false, so there is no need to call SetAsGradeintNnet().
Sometimes I declare a variable is_gradient = false just before SetZero() to document what the 'false' means.